v2/tools/generator/internal/functions/testdata/TestGolden_PropertyAssignmentFunction_WhenSharedObjectVersion/SharedObject.golden

```diff
@@ -0,0 +1,54 @@
+// Code generated by azure-service-operator-codegen. DO NOT EDIT.
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+package v20200101
+
+import (
+	personv20211231 "github.com/Azure/azure-service-operator/testing/person/v20211231"
+	personv20220630 "github.com/Azure/azure-service-operator/testing/person/v20220630"
+	"github.com/pkg/errors"
+)
+
+type Person struct {
+	Residence Location `json:"residence,omitempty"`
+}
+
+// AssignPropertiesFromPerson populates our Person from the provided source Person
+func (person *Person) AssignPropertiesFromPerson(source *personv20220630.Person) error {
+
+	// Residence
+	var locationStash personv20211231.Location
+	err := locationStash.AssignPropertiesFromLocation(&source.Residence)
+	if err != nil {
+		return errors.Wrap(err, "calling AssignPropertiesFromLocation() to populate field LocationStash from Residence")
+	}
+	var residence Location
+	err = residence.AssignPropertiesFromLocation(&locationStash)
+	if err != nil {
+		return errors.Wrap(err, "calling AssignPropertiesFromLocation() to populate field Residence from LocationStash")
+	}
+	person.Residence = residence
```

Comment:
Is it worth making this scenario have more than one intermediate version to check that the conversion generation can deal with that?

Reply:
Good idea. Done.